sdk/servicebus/service-bus/src/serviceBusMessage.ts

```diff
@@ -1085,9 +1085,10 @@ export class ServiceBusMessageImpl implements ReceivedMessageWithLock {
       throw error;
     }
     const isDeferredMessage = this._context.requestResponseLockedMessages.has(this.lockToken);
```

Comment:
The way we determine if the message was deferred is incorrect because the locktoken is removed from the `requestResponseLockedMessages` once the deferred message is settled and we have no way to know if the message is deferred from then on.

Reply:
We can use the below to throw the `Failed to settle as it is already settled` error message right?
```js
const isDeferredMessage = !this.delivery.link;
const isMessageAlreadySettled = isDeferredMessage ? !this._context.requestResponseLockedMessages.has(this.lockToken) : this.delivery.remote_settled;
```

And then use the second const above to throw the

Reply:
Was thinking the same for `isDeferredMessage` but was planning to allow the request to reach the service where it throws `The lock supplied is invalid. Either the lock expired, or the message has already been removed from the queue.`.

Reply:
Updated as suggested.

Reply:
Was wondering if
- there is a case where `this.delivery.link` is undefined for a non-deferred message
- we should let the service throw the error
- we should update `isSettled` accordingly whenever the message is settled through the management link and also expose it to the users
    ```ts
      // Currently, not exported
      public get isSettled(): boolean {
       return this.delivery.remote_settled;
      }
    ```

Reply:
When a message is "normally" received over an AMQP reciever link, there is always a "delivery" associated with it which rhea fills up. You can check to see if rhea does not do this in any scenario

For deferred messages, we create a "delivery" with only the lock token related information so that we have common code around lock renewal.